[removed]

andrewrabert opened the pull request:
[removed]

nvllsvm commented:
I'm debating where a final Qt 5 release should be made before this gets merged. Mainly to bring in the mpv 0.38.0 compatibility fix.

@iwalton3 Thoughts?

sevenrats commented:
This branch is fully unsuitable for release, and releasing it would currently add no value to the ecosystem.  Qt6 does not give us *anything* in a world where 5.15.11 exists. The most reasonable way forward is to ride Qt5 until compat loses support. Within that time we should have built an OSC per #537  or the related Qt Bug will have been fixed.  Then we can port to the native API and at that point we can discuss sunsetting the Qt5 build.  Migrating now would be way beyond premature imho.

iwalton3 commented:
This needs to be held until we either:

- reach feature parity with the 5.x branch
- are forced to switch to it because the ecosystem around it breaks

Notably the rendering pipeline is completely busted and we're using a UI cutout workaround to make it even slightly work that kills all overlay menus and trick play in the player.

iwalton3 reviewed (changes requested):
See comment.

sevenrats commented:
some of these other changes are very good @nvllsvm, do you intend to open a new pr with them?

nvllsvm commented:
@sevenrats Took a first pass and added a few of those changes to https://github.com/jellyfin/jellyfin-media-player/pull/603

satmandu commented:
Would you please add these ubuntu packages to the list of required packages for building for QT6 in Readme.md?
`qt6-5compat-dev qt6-base-dev qt6-base-private-devqt6-declarative-dev qt6-webchannel-dev qt6-webengine-dev`

kenden commented:
This would help with the message
```
Uses an end-of-life runtime
The runtime used by this app is no longer receiving security updates
```
from https://flathub.org/apps/com.github.iwalton3.jellyfin-media-player

sevenrats commented:
@kenden 
I suspect this has to do with the flatpak itself. Qt5 is not EoL for another 12 months, so, whatever runtime you have that is outdated is probably unrelated to this issue.

kenden commented:
Thanks @sevenrats I think this is actually described in this issue: https://github.com/flathub/com.github.iwalton3.jellyfin-media-player/issues/41

Pentaphon commented:
> Qt5 is not EoL for another 12 months

We are now weeks away.

iwalton3 commented:
Ahh yes the positioning workaround fix. I was never happy with it because it broke thumbnail previews and drop-down menus, hence I basically tabled this project and decided to wait until I was forced to revisit it due to Qt 5 being end of life.

I consider a proper fix to be fix rendering with upstream (unlikely) or implement a completely custom player UI.

The latter is a lot of work, and honestly may make sense to switch to a better technical baseline if we abandon the main reason we use the abomination of a rendering pipeline this uses as it would add things like proper HDR and Dolby vision support. (In other words, there's a reason the upstream Plex Media Player application passed it's sell by date and this is likely it.)

In this way MPV Shim always had a better technical foundation but the player UI leaves a lot to be desired for normal users. (I did add trick play to that though, proving you can paper over a lot of UX issues with it with enough LUA scripts though.)

Commit:
Change to Qt 6

Co-authored by: Izzie Walton <ian@iwalton.com>
Co-authored by: Satadru Pramanik <satadru@gmail.com>
Co-authored by: sevenrats <sevenrats@tutanota.com>
Co-authored by: xB <65196493+xBiei@users.noreply.github.com>
Co-authored by: zjeffer <4633209+zjeffer@users.noreply.github.com>

Commit:
Remove unused KonvergoWindow

ItsLiyua commented:
I've seen you working on this and I just wanna say thank you! Sadly I lack the skill to help. Thanks for doing this!

andrewrabert commented:
@iwalton3 This works fine on Arch Linux with the mpvqt package installed. Other platforms need fixes to get it building (eg. CI tests), but I'd like to commit any fixes to that after this PR gets merged.

Merging this would allow the Arch Linux AUR jellyfin-media-player-git package to build and be used with qt6.

I'm currently looking into the issues building on other platforms and will open PR's to fix those later.
